packages/babel-plugin-wrap-modules-amd/src/index.js

```
@@ -135,9 +135,11 @@ function applyUserDefinedTemplateIfPresent(filename, defineNode, log) {
 
 	log.info('wrap-modules-amd', 'Applied user template to wrap file');
 
-	const buildUserTemplate = template(
-		fs.readFileSync(templateFile).toString()
-	);
+	const buildUserTemplate = template(`
+		(function() {
+			${fs.readFileSync(templateFile).toString()}
```

Should we also try-catch'it?

---

:thinking: I would say no because:

1) If we catch the error and report it we are doing the same the browser does, so there's no point in doing it
2) If we catch the error and swallow it, the definition won't take place and it will fail the resolution with an obscure timeout error which is harder to debug.

---

The problem is, with combo, an error breaks the whole combo execution...

---

Yes, but still it will fail afterwards, when resolving the resolution. One combo request cannot belong to more than one `require()` resolution so, if the combo doesn't fail, the loader will, as I said.

It's what was happening in this error: the definition of two modules was done with the same name. That led to: 

1) an error in the console informing of a duplicate definition (because an exception is thrown inside the loader when a module is defined twice)
2) a following error stating a resolution timeout (the huge red message that everybody skips and forwards us :sweat_smile: )

If we had catched the double definition exception and swallowed it, it would have taken much more than one day to find the root cause. On the other hand, if we had catched and printed it, we would have been in exactly the same situation: as soon as one single module in a combo request fails to define, that `require()` is doomed.

But, if you feel it is more maintainable to catch and print to console.error (in case we reuse combo requests in the future) we can do it. However, it will impact module sizes a bit...

---

I never said `swallow` it 😉

---

But it's okay, let's push for now, since this has been the status quo for a long time

---

This is going to include some awkward whitespace now, isn't it?

---

@jbalsas I never said you said `swallow` it ;-P I just analyzed all the possibilities we had to handle the exception

@wincent Whitespace? If you mean the formatting of the IIFE, it doesn't show in the final output because it is converted into an AST prior to printing.

---

> Whitespace? If you mean the formatting of the IIFE

Yeah, that's exactly what I was referring to.

> it doesn't show in the final output because it is converted into an AST prior to printing.

Good to know. 👍